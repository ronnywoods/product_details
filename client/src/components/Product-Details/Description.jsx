import React from 'react';
import PropTypes from 'prop-types';

const Description = props => {
  return (
    <div className="container">
      <div className="row">
        <div className="col-sm shoe-desc">
          <h2 className="detail-name">{props.shoe.name}</h2>
          <h5 className="short-desc">{props.shoe.short_desc}</h5>
          <p className= "para">{props.shoe.long_desc}</p>
        </div>
        <div className="col-sm shoe-img">
          <img src={props.shoe.img_url}></img>
        </div>
      </div>
    </div>
  )
}

Description.propTypes = {
  shoe: PropTypes.object,
};

export default Description;
