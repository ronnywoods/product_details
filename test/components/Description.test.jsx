import React from 'react';
import { shallow } from 'enzyme';

import Description from '../../client/src/components/Product-Details/Description';

describe('Description Component', () => {
  const desc = { short_desc: 'This is a short description',
    name: 'ULTRABOOST',
    long_desc: 'This is a long description of ultraboost.',
    img_url: 'https://assets.adidas.com/images/w_840,h_840,f_auto,q_auto:sensitive,fl_lossy/29ffefeadfb448bfae6ba93b007c3f38_9366/POD-S3_1_Shoes_Black_G54741_04_standard.jpg',
  };
  const component = shallow(<Description shoe={desc} />);

  it('should render description correctly when passed in descriptions', () => {
    expect(component).toMatchSnapshot();
  });

  it('should display shoe name properly', () => {
    expect(component.find('h2.detail-name').text()).toBe('ULTRABOOST');
  });

  it('should display short description properly', () => {
    expect(component.find('h5.short-desc').text()).toBe('This is a short description');
  });

  it('should display long description properly', () => {
    expect(component.find('p.para').text()).toBe('This is a long description of ultraboost.');
  });

  it('should have img of shoe', () => {
    expect(component.find('img')).toHaveLength(1);
  });
});
