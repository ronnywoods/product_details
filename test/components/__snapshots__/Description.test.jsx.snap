// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description Component should render description correctly when passed in descriptions 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-sm shoe-desc"
    >
      <h2
        className="detail-name"
      >
        ULTRABOOST
      </h2>
      <h5
        className="short-desc"
      >
        This is a short description
      </h5>
      <p
        className="para"
      >
        This is a long description of ultraboost.
      </p>
    </div>
    <div
      className="col-sm shoe-img"
    >
      <img
        src="https://assets.adidas.com/images/w_840,h_840,f_auto,q_auto:sensitive,fl_lossy/29ffefeadfb448bfae6ba93b007c3f38_9366/POD-S3_1_Shoes_Black_G54741_04_standard.jpg"
      />
    </div>
  </div>
</div>
`;
